### PR TITLE
fixed case of mapping merge not working across files

### PR DESCRIPTION
**Pull request by joedevivo**

### CODE REVIEW

File: src/cuttlefish_schema.erl [hunk 1]
```diff
@@ -49,14 +49,14 @@ merger(Fun, ListOfInputs) ->
...
-            case Fun(Input) of
+            case Fun(Input, {TranslationAcc, MappingAcc, ValidatorAcc}) of
                 {error, Errors} ->
                     %% These have already been logged. We're not moving forward with this
                     %% but, return them anyway so the rebar plugin can display them
                     %% with io:format, since it doesn't have lager.
                     {error, Errors};
                 {Translations, Mappings, Validators} ->
-
+%                    {Translations, Mappings, Validators}
```
Comment (seancribbs) [inline src/cuttlefish_schema.erl]: Stray comment?

Comment (joedevivo) [inline src/cuttlefish_schema.erl]: yup.

File: src/cuttlefish_schema.erl [hunk 2]
```diff
@@ -149,8 +149,8 @@ string(S) ->
             {error, [{error, [ErrStr]}]}
     end.
 
-parse_schema(Tokens, Comments) ->
-    parse_schema(Tokens, Comments, {[], [], [], []}).
+%parse_schema(Tokens, Comments) ->
```
Comment (seancribbs) [inline src/cuttlefish_schema.erl]: Lingering removed function?

Comment (joedevivo) [inline src/cuttlefish_schema.erl]: yup

**Comment by seancribbs** [comment]
:+1: This definitely feels better in terms of structure and seems to solve the merge issue.

File: src/cuttlefish_schema.erl [hunk 3]
```diff
@@ -49,14 +49,13 @@ merger(Fun, ListOfInputs) ->
     Schema = lists:foldr(
         fun(Input, {TranslationAcc, MappingAcc, ValidatorAcc}) ->
 
-            case Fun(Input) of
+            case Fun(Input, {TranslationAcc, MappingAcc, ValidatorAcc}) of
                 {error, Errors} ->
                     %% These have already been logged. We're not moving forward with this
                     %% but, return them anyway so the rebar plugin can display them
                     %% with io:format, since it doesn't have lager.
                     {error, Errors};
```
Comment (seancribbs) [inline src/cuttlefish_schema.erl:57]: Won't this cause a `function_clause` error on the next iteration?

**Pull request merged**